### ISSUE TITLE
Does this work with zfs? 

### ISSUE THREAD

**Issue by stevezau** [ckcr4lyf/qbit-race#35]
Hi,

I tried to run

qbit-race add -p "/data/downloads/somedownload"

But i get the following error
```2023-05-28T05:25:00.118Z [ERR]: Failed to read torrent from Error: EISDIR: illegal operation on a directory, read```

I've checked the permissions etc.. Any idea what could be causing it or how to t-shoot?

**Comment by ckcr4lyf** [comment]
It does not support passing a path to a whole directory, but rather a single torrent file, e.g. `/data/downloads/somedownload/linux_iso_1080p.torrent`

Is this what you're trying to do, or trying to add _all_ `.torrent` files from a specified directory?

**Comment by stevezau** [comment]
Ah, obviously i have no idea how this tool works lol. 

In my setup, autobrr sends to sonarr/radarr, then they fw to qbit. I thought this tool could race with an existing torrent already loaded into qbittorrent but it seems it actually loads it, is that right?

So i'd need to get sonarr/radarr to call qbit-race?

**Comment by ckcr4lyf** [comment]
Yes, the tool is meant to load the torrents directly into qbittorrent, especially to do stuff like re-announce and apply the rules at load time to optimize for racing performance. 

If autobrr sends them to *arr, which then sends them to qBittorrent, you might need to check if you can customize how the *arr adds it (e.g. by executing a custom command). 

I've not used *arrs before, but I'll give it a look. If you use direct autobrr -> qBittorrent it is definitely possible.

**Comment by stevezau** [comment]
Ah, now it makes sense.

I depend on the arr apps to determine if I want a release. i.e. is it a show/movie in my library, is it a better quality than my existing one, and is the file size within specific parameters? 

I could use a "torrent blackhole" in the arr apps, which allows you to execute a script, but you lose its management. 

So it seems maybe this tool isn't a good fit for my use cases. Unless it be modified to also support finding a torrent in an existing qubit instance instead of loading it?

**Comment by ckcr4lyf** [comment]
That's definitely possible, something like the infohash would work best. But how would you trigger it? Do the arrs have a post action script?

**Comment by stevezau** [comment]
I think an infohash would be possible.. I think there could be a number of ways to do it.. off the top of my head..

option 1
we can call a post-action script in autobrr.. that script can find the infohash then send it to qbit-race. The script might need to wait a few seconds for it to show up in qbit tho.

option2 (better IMO)
sonarr/radarr have event triggers https://wiki.servarr.com/radarr/settings#connections which you can do various things on incl call a script. I believe the event will contain data of the action. I just ran a test on the "on grab" action. Here is the payload

It looks like the downloadId contains the info hash.. So it should be possible?  

```
{
  "movie": {
    "id": 17706,
    "title": "xxxx",
    "year": 2023,
    "releaseDate": "9999-12-31",
    "folderPath": "/data/Movies/xxxxx",
    "tmdbId": xxxxx,
    "imdbId": "xxxxxx",
    "overview": "xxxxx"
  },
  "remoteMovie": {
    "tmdbId": xxxx,
    "imdbId": "xxxx",
    "title": "xxxx",
    "year": 2023
  },
  "release": {
    "quality": "WEBDL-1080p",
    "qualityVersion": 1,
    "releaseGroup": "QuaSO",
    "releaseTitle": "xxxxxxx 1080p WEB-DL DD+5.1 H.264-QuaSO",
    "indexer": "Tracker1",
    "size": 5173933941,
    "customFormatScore": 1750,
    "customFormats": [
      "DD+",
      "x264"
    ],
    "indexerFlags": []
  },
  "downloadClient": "qbit",
  "downloadClientType": "qBittorrent",
  "downloadId": "BAFC795A07FEE0F6355299F769E592EE39F5D4D6",
  "customFormatInfo": {
    "customFormats": [
      {
        "id": 104,
        "name": "DD+"
      },
      {
        "id": 138,
        "name": "x264"
      }
    ],
    "customFormatScore": 1750
  },
  "eventType": "Grab",
  "instanceName": "Radarr",
  "applicationUrl": ""
}```

**Comment by ckcr4lyf** [comment]
So would you be able to configure sonarr to, once it makes a decision and adds the torrent, call a program e.g. `qbit-race -i [infohash]`?

If so I can work on adding this feature, to manually race an existing torrent based on infohash.

**Comment by stevezau** [comment]
Yes, you configure sonarr/radarr to call a script post it adding to qbit. I would need to write a little bash script to extract the infohash from the payload and pass it to qbit-race.

**Comment by stevezau** [comment]
If you can get it working, i will write the script.. I could also open a PR to your readme.md with the sample script and instructions.
